### DISCUSSION TITLE
Force container to a workspace

### DISCUSSION THREAD

**Discussion by LarveyOfficial** [zen-browser/desktop#1471] [Ideas]
Would be very nice if we could enable a setting (opt-in) that would force a specific container to be automatically moved into a workspace. So if I set a workspace to a "Work" container, and I open a link in my personal container that I have set to always open in a specific container, or if I have a shortcut that opens a new "Work" container tab, it would automatically move to the "Work" workspace and automatically switch me over to it.

This setting could be a simple check mark that appears under the "Set Default Container" option if and only if you have a container set that isn't "No container".

**Comment by ju-li** [comment]
Upvote this idea: https://github.com/zen-browser/desktop/discussions/887

It's pretty much the same thing!

**Reply by LarveyOfficial** [reply]
This has already been technically implemented. You can set a default container per each tab, but you can't force a workspace to only be able to contain 1 type of container.

Right now you can set a default container, but lets say I have workspace "Work" with default container "Work" set, then I move to workspace "Personal" and open a "Work" container tab either via going to a website that is set to always open in "Work" or via using a shortcut, then it opens that container in the "Personal" workspace. I'm proposing is that when you do open that "Work" container in the "Personal" workspace, it instead forces that new tab to go to the "Work" Workspace. This is similar to how sideberry lets you redirect containers to specific pages.

**Comment by SJ50** [comment]
How to add more containers?

**Reply by Truleu** [reply]
Try installing the Firefox Multi-Account Containers extension. It gives more options to control the containers

**Reply by SJ50** [reply]
@mauro-balades is this a correct way of adding more containers?

**Reply by LarveyOfficial** [reply]
Yes

**Comment by jackperry** [comment]
Adding another "please" to this one! Default container per Workspace is really useful, but when combined with Multi-Account Containers and websites forced to those containers, opening a website forced to a container does work, but it opens that container tab in the default workspace. It would be incredibly helpful if Zen could intercept that container tab and basically perform the "Change Tab(s) to Workspace" right-click feature for me automatically.

**Comment by hktr92** [comment]
It would be awesome if every workspace would have it's own container "by default"!

**Comment by far19a** [comment]


**Reply by far19a** [reply]
I love Firefox' container, it's great for separating web user profile without creating new browser profile.

**Comment by kristijanribaric** [comment]
Implemented with this PR: https://github.com/zen-browser/desktop/pull/2095
